Samples/StyleTransfer/AppViewModel.cs

```diff
@@ -0,0 +1,546 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Windows.Input;
+using Windows.Media.Core;
+using GalaSoft.MvvmLight.Command;
+using Windows.Media.Capture.Frames;
+using System.Diagnostics;
+using Windows.Media.Capture;
+using Windows.Media.Playback;
+using Windows.Media.Effects;
+using Windows.Media;
+using Windows.Foundation.Collections;
+using Microsoft.AI.MachineLearning;
+using Windows.Storage;
+using System.Runtime.CompilerServices;
+using Windows.Graphics.Imaging;
+using Windows.UI.Xaml.Media.Imaging;
+using Windows.UI.Xaml.Media;
+using Windows.Media.MediaProperties;
+using Windows.Storage.Pickers;
+using Windows.UI.Xaml.Controls;
+using System.IO;
+using Windows.System.Display;
+using Windows.UI.Core;
+using GalaSoft.MvvmLight.Threading;
+using System.Threading;
+using StyleTransferEffectCpp;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Windows.Devices.Enumeration;
+
+namespace StyleTransfer
+{
+    class AppViewModel : INotifyPropertyChanged
+    {
+        public AppViewModel()
+        {
+            _appModel = new AppModel();
+            SaveCommand = new RelayCommand(SaveOutput);
+            SetMediaSourceCommand = new RelayCommand<string>(async (x) => await SetMediaSource(x));

```

Comment:
>SetMediaSource [](start = 80, length = 14)

looking at this it looks like the viewmodel should just update the underlying model here: 

_appModel.InputMedia = src;

After looking closely, it looks like the xaml should probably bind directly to InputMedia as an enum, when that property is updated, then viewmodel should respond to the changes in the model by listening to the propertychanged callback from the model. 

A normal property changed callback should have the sender, which can be cast to the viewmodel, to query for additional properties on the model without relying on member variables.